homeassistant/components/melcloud/climate.py

```diff
@@ -164,10 +179,13 @@ def extra_state_attributes(self) -> dict[str, Any] | None:
     @property
     def hvac_mode(self) -> str:
         """Return hvac operation ie. heat, cool mode."""
-        mode = self._device.operation_mode
-        if not self._device.power or mode is None:
-            return HVAC_MODE_OFF
-        return ATA_HVAC_MODE_LOOKUP.get(mode)
+        op_mode = self._device.operation_mode
+        if not self._device.power or op_mode is None:
+            mode = HVAC_MODE_OFF
+        else:
+            mode = ATA_HVAC_MODE_LOOKUP.get(op_mode)
+        self._attr_hvac_action = HA_STATE_TO_CURRENT_HVAC.get(mode)
```

Comment:
This is adding simulation / assumed state. The action is used to what it is doing vs what it is set to. However, this change does not make that difference.

Reply:
Adding this just because there are some integration (e.g. HomeKit) that use HVAC ACTION to determinate and reflect the status (in detail the icon color) of associated devices. I understand the difference between ACTION and MODE, but inspecting the code I see that many other climate integration implement also the ACTION just reflecting the MODE so I thought that is should be useful. I can close this PR if you think that this is wrong.

Reply:
I think that is a HomeKit integration thing to solve in that case? Feel a bit weird to add something that is not true to fix HomeKit

Reply:
Understand your point, I was just following behavior from other climate integration. Let me just say that information provided is not wrong, e.g. when `Melcloud` climate is in `Cool` mode for sure is `Cooling` (at list for my devices do not exist an `Idle` state). So should we close?

Reply:
Let's get a second opinion on this.

Reply:
Waiting for second opinion, I think about this on this days and I review behavior of others climate entities. At the end only Daikin do something similar, and anyway try to identify an `idle` state if the model support this information. Probably provide this property as pure duplicate of HVAC mode does not make sense, may be HomeKit integration should be fixed to use HVAC mode when HVAC action is not available.